sys/net/gnrc/netif/lorawan/gnrc_netif_lorawan.c

```diff
@@ -69,7 +70,14 @@ void gnrc_lorawan_mlme_confirm(gnrc_lorawan_t *mac, mlme_confirm_t *confirm)
 
     if (confirm->type == MLME_JOIN) {
         if (confirm->status == 0) {
-            DEBUG("gnrc_lorawan: join succeeded\n");
+            gnrc_netif_lorawan_t *lw_netif = container_of(mac, gnrc_netif_lorawan_t, mac);
+            gnrc_netif_t *netif = container_of(lw_netif, gnrc_netif_t, lorawan);
+
+            printf("gnrc_lorawan: join succeeded %d\n", netif->pid);
+            if (netif->dev) {
+                netif->lorawan.flags |= GNRC_NETIF_LORAWAN_FLAGS_JOINED;
```

Comment:
this flag seems to be write only. Do you plan to use it later?

Reply:
Yes, for the IID getter I plan to use it: If the interface is connected, return the IID, but -ENOTCONN otherwise. This way the error check introduced in this PR for getting the IID in the NIB, will also guarantee, that the device does not get an address for an empty IID. Using a flag is far less complicated than checking the UP status via MLME (as the structs require nearly 100 bytes on the stack if used).